grove/alpha/simon/simon.py

```diff
@@ -13,441 +13,334 @@
-    :math:`\\mathbf{\\mathit{W}}` is return unchanged.
-
-    :param 2darray W: a matrix of 0s and 1s in row-echelon form, with rows all
-                      orthogonal to some vector of 0s and 1s.
-
-    :param 1darray z: a vector of 0s and 1s known
-                      to be orthogonal to :math:`\\mathbf{s}`
-    :return: either the same matrix :math:`\\mathbf{\\mathit{W}}`, unchanged,
-             or :math:`\\mathbf{\\mathit{W}}` with one additional
-             row added that maintains the property described above.
-    :rtype: 2darray
-    """
-    n = len(z)
-    while np.any(z != 0):  # while z is not all zeros
-        if len(W) == 0:
-            W = z
-            W = W.reshape(1, n)
-            break
-        msb_z = most_significant_bit(z)
-
-        # Search for a row to insert z into,
-        # so that it has an earlier significant bit than the row below
-        # and a later one than the row above (when reading left-to-right)
-        got_to_end = True
-        for row_num in range(len(W)):
-            row = W[row_num]
-            msb_row = most_significant_bit(row)
-            # if the row as the same msb as z,
-            # set z to the bitwise xor of z and the current row
-            # as it will be guaranteed to still be orthogonal to s
-            if msb_row == msb_z:
-                z = np.array([z[i] ^ row[i] for i in range(n)])
-                got_to_end = False
-                break
-            # if the row has a greater msb than z,
-            # then this is the row to z insert above
-            elif msb_row > msb_z:
-                W = np.insert(W, row_num, z, 0)
-                got_to_end = False
-                break
-        # if z has a greater msb than all rows,
-        # insert it to the bottom of the array
-        if got_to_end:
-            W = np.insert(W, len(W), z, 0)
-
-    return W
-
-
-def make_square_row_echelon(W):
-    """
-    Make :math:`\\mathbf{\\mathit{W}}`
-    into a square matrix for Simon's algorithm, satisfying a few criteria.
-
-    :param 2darray W: an :math:`(n-1)\\times n` array of 0s and 1s in
-                      row-echelon form such that all rows are orthogonal
-                      to some length :math:`n` vector
-                      of 0s and 1s :math:`\\mathbf{s}`.
-    :return: a two-element tuple. The first element is
-             an :math:`n\\times n` square array identical
-             to :math:`\\mathbf{\\mathit{W}}` except with one row added.
-             That row is chosen to keep :math:`\\mathbf{\\mathit{W}}`
-             in row-echelon form. The second element is the row that
-             the new row is in, where the top row is at index 0.
-    :rtype: tuple
-    """
-    n = len(W) + 1
+    periodic_bit_map_dct = {}
+    for cnt in range(n_samples):
+        bitstring_tup = list_of_bitstring_tuples[cnt]
+        val = range_of_2to1_map[cnt]
+        periodic_bit_map_dct[bitstring_tup[0]] = val
```

Comment:
Is there something periodic about *this* bit map?

Reply:
yes, because we create a period 2 bit map in this method

Reply:
removed the `perodic` to avoid confusion